VideoLocker/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -70,18 +71,27 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                     "ALTER TABLE "    + DbStructure.Table.DOWNLOADS + " ADD COLUMN "
                                       + DbStructure.Column.URL_YOUTUBE + " TEXT "};
 
-            if (oldVersion == 1 && (newVersion == 2 || newVersion == 3)) {
+            String upgradeToV4 =
+                    "ALTER TABLE "    + DbStructure.Table.DOWNLOADS + " ADD COLUMN "
+                            + DbStructure.Column.VIDEO_FOR_WEB_ONLY + " BOOLEAN ";
+
+            if ( oldVersion == 1 ) {
                 // upgrade from 1 to 2
                 db.execSQL(upgradeToV2);
             }
 
-            if (newVersion == 3) {
+            if ( oldVersion < 3 ) {
```

Comment:
@BenjiLee  I changed the code to make the logic for progressive upgrading simple